gauntlet/packages/gauntlet-serum-multisig/src/commands/multisig.ts

```diff
@@ -132,71 +144,114 @@ export const wrapCommand = (command) => {
+          pubkey: this.multisigAddress,
+          isWritable: false,
+          isSigner: false,
         },
-      )
-      return tx
+        {
+          pubkey: proposal,
+          isWritable: true,
+          isSigner: false,
+        },
+        {
+          pubkey: this.wallet.payer.publicKey,
+          isWritable: false,
+          isSigner: true,
+        },
+        {
+          pubkey: SYSVAR_RENT_PUBKEY,
+          isWritable: false,
+          isSigner: false,
+        },
+      ]
+      const rawTx: RawTransaction = {
+        data,
+        accounts,
+        programId: this.program.programId,
+      }
+      return [rawTx]
     }
 
-    approveProposal: ProposalAction = async (proposal: PublicKey): Promise<string> => {
-      logger.loading(`Approving proposal for ${command.id}`)
-      const tx = await this.program.rpc.approve({
-        accounts: {
-          multisig: this.multisigAddress,
-          transaction: proposal,
-          owner: this.wallet.publicKey,
+    approveProposal: ProposalAction = async (proposal: PublicKey): Promise<RawTransaction[]> => {
+      logger.loading(`Generating proposal approval data for ${command.id}`)
+      const data = this.program.coder.instruction.encode('approve', {})
+      const accounts: AccountMeta[] = [
+        {
+          pubkey: this.multisigAddress,
+          isWritable: false,
+          isSigner: false,
         },
-      })
-      return tx
+        {
+          pubkey: proposal,
+          isWritable: true,
+          isSigner: false,
+        },
+        {
+          pubkey: this.wallet.publicKey,
+          isWritable: false,
+          isSigner: true,
+        },
+      ]
+      const rawTx: RawTransaction = {
+        data,
+        accounts,
+        programId: this.program.programId,
+      }
+      return [rawTx]
     }
 
-    executeProposal: ProposalAction = async (proposal: PublicKey, context): Promise<string> => {
-      logger.loading(`Executing proposal for ${command.id}`)
-      // get the command's starting word to map it to the respective IDL(ocr2, store etc)
-      const { idl } = getContract(command.id.split(':')[0], '')
-      try {
-        const tx = await this.program.rpc.executeTransaction({
-          accounts: {
-            multisig: this.multisigAddress,
-            multisigSigner: context.multisigSigner,
-            transaction: proposal,
-          },
-          remainingAccounts: context.proposalState.accounts
-            .map((t) => (t.pubkey.equals(context.multisigSigner) ? { ...t, isSigner: false } : t))
-            .concat({
-              pubkey: context.proposalState.programId,
-              isWritable: false,
-              isSigner: false,
-            }),
+    executeProposal: ProposalAction = async (proposal: PublicKey, context): Promise<RawTransaction[]> => {
+      logger.loading(`Generating proposal execution data for ${command.id}`)
+      const data = this.program.coder.instruction.encode('executeTransaction', {})
```

Comment:
Why the change here?

Reply:
We don't want to execute the proposal directly, but generate the tx and decide based on the user instructions